modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -202,6 +210,243 @@ TEST(GPU, Symm7x7_test)
+    }
+    if (*device == nullptr) {
+        CV_LOG_DEBUG(NULL, "OpenCL: D3D11CreateDevice returns empty ID3D11Device\n");
+        throw SkipTestException("D3D11CreateDevice was failed");
+    }
+}
+
+void prepareD3D11Texture2D(const UINT width, const UINT height, std::vector<uint8_t>& test_data,
+                           ID3D11Texture2D** texture,
+                           ID3D11Device** device) {
+    // Create device
+    *device = nullptr;
+    createD3D11Device(device);
+
+    // Create texture description
+    D3D11_TEXTURE2D_DESC desc = {};
+    desc.Width = width;
+    desc.Height = height;
+    desc.MipLevels = 1;
+    desc.ArraySize = 1;
+    desc.Format = DXGI_FORMAT_A8_UNORM; // CV_8U
+    desc.SampleDesc.Count = 1;
+    desc.Usage = D3D11_USAGE_DEFAULT;
+    desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
+    desc.MiscFlags = D3D11_RESOURCE_MISC_SHARED;
+    desc.CPUAccessFlags = 0; // 0 if CPU access is not required
+
+    // Associate subresource with test data
+    std::iota(test_data.begin(), test_data.end(), uint8_t(0));
+    D3D11_SUBRESOURCE_DATA initData = { test_data.data(),
+                                        sizeof(uint8_t) * width,
+                                        height };
+
+    // Create texture2D by test data
+    HRESULT hr = (*device)->CreateTexture2D(&desc, &initData, &(*texture));
+
+    if(FAILED(hr)) {
+        CV_LOG_DEBUG(NULL, "CreateTexture2D was failed: " << std::to_string(HRESULT_CODE(hr)));
+        throw SkipTestException("CreateTexture2D was failed");
+    }
+
+    if(texture == nullptr) {
+        throw SkipTestException("CreateTexture2D was failed: texture ptr is NULL");
+    }
+}
+
+int findPlaceOfDevice(const std::vector<std::pair<bool, cv::ocl::Device>>& devices,
+                      bool preffered = true) {
+    auto plc = std::find_if(devices.begin(), devices.end(),
+                            [&preffered](std::pair<bool, cv::ocl::Device> cl_device) {
+                                if (!cl_device.first && !preffered) return false;
+                                if (cl_device.second.isIntel()) return true;
+                                else return false;
+                            });
+    if(plc != devices.end()) return int(std::distance(devices.begin(), plc));
+    else return -1;
+}
+
+TEST(GPU_D3D11, getDeviceIDs)
+{
+    // Create test data ///////////////////////////////////////////////////////
+    ID3D11Device* pD3D11Device = nullptr;
+    createD3D11Device(&pD3D11Device);
+    std::vector<std::pair<bool, cv::ocl::Device>> devices;
+
+    // Get Device IDs for all platforms //////////////////////////////////////
+    EXPECT_NO_THROW(cv::directx::getDeviceIDsByD3D11Device(pD3D11Device, devices));
+
+    // Check  //////////////////////////////////////
+    EXPECT_FALSE(devices.empty());
+
+    int device_is = -1;
+    const bool prefered = true;
+    device_is = findPlaceOfDevice(devices, prefered);
```

Comment:
please put simple check for the first element true or false

Reply:
Added